lib/pure/collections/deques.nim

```diff
@@ -470,6 +470,42 @@ proc popLast*[T](deq: var Deque[T]): T {.inline, discardable.} =
   result = deq.data[deq.tail]
   destroy(deq.data[deq.tail])
 
+proc rotateLeft*[T](deq: var Deque[T], n = 1) {.since: (1, 5, 1).} =
+    ## Moves first item in the deque to the back `n` number of times.
+    ## Equivalent (but faster) than `deq.addLast(deq.popFirst())` 
+    runnableExamples:
+      var a = toDeque([1, 2, 3, 4, 5])
```

Comment:
add a more comprehensive test to tdeques.nim

Reply:
going to do that now, thanks

Reply:
sorry i'm sort of a novice with git, how can i sync my fork with the latest update on the main nim repo so i can edit that tdeques file?

Reply:
nvm, got it. i added the since notation on rotateRight and i'm going to do the tests now.